lib/nav.py

```diff
@@ -10,7 +10,72 @@ class DIRS(Enum):
 class OPS(Enum):
 	DIG='DIG'
 	MOVE='MOVE'
+	TURN='TURN'
+        
+class TURNS(Enum):
+	LEFT='LEFT'
+	RIGHT='RIGHT'
 
+class AXIS(IntEnum):
+	X=0
+	Y=1
+	Z=2
+
+class CARDINALS(Enum):
+	NORTH='NORTH'
+	EAST='EAST'
+	SOUTH='SOUTH'
+	WEST='WEST'
+
+# maps a direction to an axis
+DIRS_TO_AXIS = {
+	DIRS.UP: {
+		'AXIS':AXIS.Z,
+		'DIRECTION':1,
+	},
+	DIRS.DOWN: {
+		'AXIS':AXIS.Z,
+		'DIRECTION':-1,
+	},
+	DIRS.FORWARD: {
+		CARDINALS.NORTH: {
+			'AXIS':AXIS.Y,
+			'DIRECTION':1,
+		},
+		CARDINALS.EAST: {
+			'AXIS':AXIS.X,
+			'DIRECTION':1,
+		},
+		CARDINALS.SOUTH: {
+			'AXIS':AXIS.Y,
+			'DIRECTION':-1,
+		},
+		CARDINALS.WEST: {
+			'AXIS':AXIS.X,
+			'DIRECTION':-1,
+		},
+	},
+}
+
+# maps cardinal direction and a turn to a resulting cardinal direction
+TURN_TO_DIR = {
+	CARDINALS.NORTH: {
+		TURNS.LEFT:CARDINALS.WEST,
+		TURNS.RIGHT:CARDINALS.EAST,
+	},
+	CARDINALS.EAST: {
+		TURNS.LEFT:CARDINALS.NORTH,
+		TURNS.RIGHT:CARDINALS.SOUTH,
+	},
+	CARDINALS.SOUTH: {
+		TURNS.LEFT:CARDINALS.EAST,
+		TURNS.RIGHT:CARDINALS.WEST,
+	},
+	CARDINALS.WEST: {
+		TURNS.LEFT:CARDINALS.SOUTH,
+		TURNS.RIGHT:CARDINALS.NORTH,
+	},
+}
```

Comment:
Couldn't you just use assign north east south and west to 0 1 2 and 3 respectively and do modular arithmetic?

Like if 0 is north, turning right is 1 and left is -1

That's a standard way to handle navigation with cardinal directions without hardcoding

For example, turtle is facing east (val of 1), he wants to face south (val of 2). 

south - east => 2 - 1 => -1, so he needs one right turn

You could get rid of lines 19-78 like this

Reply:
Will do thank you